pkg/forwarder/forwarder.go

```diff
@@ -0,0 +1,65 @@
+package forwarder
+
+import (
+	"context"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type Forwarder interface {
+	Start(context.Context) error
+}
+
+type forwarder struct {
+	srcAddress           string
+	srcPort              int
+	dstAddress           string
+	dstPort              int
+	ipvs                 bool
+	ipvsForwardingMethod string
+}
+
+func NewForwarder(srcAddress string, srcPort int, dstAddress string, dstPort int, ipvs bool, ipvsForwardingMethod string) Forwarder {
+	return &forwarder{
+		srcAddress:           srcAddress,
+		srcPort:              srcPort,
+		dstAddress:           dstAddress,
+		dstPort:              dstPort,
+		ipvs:                 ipvs,
+		ipvsForwardingMethod: ipvsForwardingMethod,
+	}
+}
+
+func (f *forwarder) Start(ctx context.Context) error {
+	log.Infoln("starting forwarder")
+	// If backend address is quad-zero route, early return
+	// (destination address is unknown, can't setup forwarding).
+	// If at some point we want to prepare forwarding none the less we would have to pick up any interface
+	// address and continue with it. If the user want to profit from different vip port and backend port
+	// it's probably easier the user define the backend address instead
+	if f.dstAddress == "0.0.0.0" {
```

Comment:
Do we need to check for ::/0 on ipv6 as well or will this only work on ipv4?

Reply:
As it is it'll only works with ipv4 for iptables variant and probably it'll be fine with ipvs (at least, as much as the current load balancer used by kube-vip).
Could we address the ipv6 support in another contribution ? maybe we could here detect ipv6+iptables forwarding and then trigger a fatal with a message to inform the user the combo issn't yet supported ?

Reply:
Sure probably worth adding a TODO and make users aware of potential limitations

Reply:
Done: https://github.com/kube-vip/kube-vip/pull/333/files#diff-c42cd718bf633c200838c1ba185053d11240419f8646f960f7c8ab7d6d8ccdb4R14